cpp/src/copying/contiguous_split.cu

```diff
@@ -72,62 +112,183 @@ void copy_in_place_kernel( column_device_view const in,
       // next grid
       warp_cur += warps_per_grid;
       index += block_size * gridDim.x;
-   }      
+   }
+}
+
+/**
+ * @brief Copies contents of one string column to another.  Copies validity if present
+ * but does not compute null count.  
+ * 
+ * This purpose of this kernel as a standlone is to reduce the number of
+ * kernel calls for copying a string column from 2 to 1, since number of kernel calls is the
+ * dominant factor in large scale contiguous_split() calls.  To do this, the kernel is
+ * invoked with using max(num_chars, num_offsets) threads and then doing seperate 
+ * bounds checking on offset, chars and validity indices.
+ * 
+ * Also handles the case unique to contiguous_split() where
+ * the offsets in the output column need to be shifted by the position of the split. This
+ * happens because the output columns from contiguous_split() are full copies of the 
+ * source columns.  So if we had a 4 element string column of the form  
+ * 
+ * chars   { "a", "b", "c", "d" }
+ * offsets {  0,   1,   2,   3,   4}
+ * 
+ * and we did a traditional split() on it in the middle, we would get two output columns 
+ * pointing to the same in-memory chars and offsets data.
+ * 
+ * chars   { "a", "b" }
```

Comment:
That's technically true, but that's an under-the-hood detail.  The { c, d } column (call it X) starts at the actual base address of the { a, b, c, d } column but from the end user's perspective it's still just a 2 element string column starting with "c".  It just so happens that in the context of the string_view, it returns you element 0 of the column by using an offset of 2.
```
X.element(0) == X.chars[X.offsets[0]] == 'c'
```
But in the contiguous_split case, all the arrays are fresh pointers, so "c" does indeed start at offset 0 relative to it's base ptr.

Reply:
But the example shows `chars   { "c", "d" }`, which has two elements, and `offsets[0] == 2`, which is out of bounds.

Reply:
Yes, but that's the point of the example.   For a traditional sliced string column that would be legit.  You would have the strings { "c", "d" }  but the offset for "c" would 2 because under the hood the base pointer for that column would really be "a".  The user just doesn't see that.

For contiguous_split all the columns have their own base address so the offset needs to be shifted down.

Reply:
I'm sorry, I was reading this as being an actual example of contiguous_split. I didn't realize you were giving an example of what the function "shouldn't" do. I think it would be nice to have the docs just explain what the function does.

Reply:
Updated/simplified the comments.